### ISSUE TITLE
Fix reduction dim order in `PropagateLinalgTransposePass`

### ISSUE THREAD

**Issue by nirvedhmeshram** [iree-org/iree#23196]
```
func.func @main(%arg0: !torch.tensor<[16,48,1,32,288],bf16>, %arg1: !torch.vtensor<[288,2,288,1,1],bf16>, %arg2: !torch.vtensor<[16,2,48,32,288],bf16>) attributes {torch.assume_strict_symbolic_shapes} {
  %none = torch.constant.none
  %false = torch.constant.bool false
  %0 = torch.prim.ListConstruct  : () -> !torch.list<int>
  %int1 = torch.constant.int 1
  %int2 = torch.constant.int 2
  %int1_0 = torch.constant.int 1
  %int1_1 = torch.constant.int 1
  %1 = torch.prim.ListConstruct %int2, %int1_0, %int1_1 : (!torch.int, !torch.int, !torch.int) -> !torch.list<int>
  %int0 = torch.constant.int 0
  %int0_2 = torch.constant.int 0
  %int0_3 = torch.constant.int 0
  %2 = torch.prim.ListConstruct %int0, %int0_2, %int0_3 : (!torch.int, !torch.int, !torch.int) -> !torch.list<int>
  %int1_4 = torch.constant.int 1
  %int1_5 = torch.constant.int 1
  %int1_6 = torch.constant.int 1
  %3 = torch.prim.ListConstruct %int1_4, %int1_5, %int1_6 : (!torch.int, !torch.int, !torch.int) -> !torch.list<int>
  %int0_7 = torch.constant.int 0
  %int4 = torch.constant.int 4
  %int1_8 = torch.constant.int 1
  %int2_9 = torch.constant.int 2
  %int3 = torch.constant.int 3
  %4 = torch.prim.ListConstruct %int0_7, %int4, %int1_8, %int2_9, %int3 : (!torch.int, !torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
  %5 = torch.aten.permute %arg2, %4 : !torch.vtensor<[16,2,48,32,288],bf16>, !torch.list<int> -> !torch.vtensor<[16,288,2,48,32],bf16>
  %int0_10 = torch.constant.int 0
  %int2_11 = torch.constant.int 2
  %int1_12 = torch.constant.int 1
  %int3_13 = torch.constant.int 3
  %int4_14 = torch.constant.int 4
  %6 = torch.prim.ListConstruct %int0_10, %int2_11, %int1_12, %int3_13, %int4_14 : (!torch.int, !torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
  %7 = torch.aten.permute %arg1, %6 : !torch.vtensor<[288,2,288,1,1],bf16>, !torch.list<int> -> !torch.vtensor<[288,288,2,1,1],bf16>
  %8 = torch.aten.convolution %5, %7, %none, %1, %2, %3, %false, %0, %int1 : !torch.vtensor<[16,288,2,48,32],bf16>, !torch.vtensor<[288,288,2,1,1],bf16>, !torch.none, !torch.list<int>, !torch.list<int>, !torch.list<int>, !torch.bool, !torch.list<int>, !torch.int -> !torch.vtensor<[16,288,1,48,32],bf16>
  %int0_15 = torch.constant.int 0
  %int3_16 = torch.constant.int 3
  %int2_17 = torch.constant.int 2
  %int4_18 = torch.constant.int 4
  %int1_19 = torch.constant.int 1
  %9 = torch.prim.ListConstruct %int0_15, %int3_16, %int2_17, %int4_18, %int1_19 : (!torch.int, !torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
  %10 = torch.aten.permute %8, %9 : !torch.vtensor<[16,288,1,48,32],bf16>, !torch.list<int> -> !torch.vtensor<[16,48,1,32,288],bf16>
  torch.overwrite.tensor.contents %10 overwrites %arg0 : !torch.vtensor<[16,48,1,32,288],bf16>, !torch.tensor<[16,48,1,32,288],bf16>
  return
}
```
For this input we end with a swapped reduction dim order that causes codegen perf issue

**Comment by IanWood1** [comment]
@nirvedhmeshram I opened https://github.com/iree-org/iree/pull/23489 to fix the problem.
